packages/core/src/modules/credentials/formats/jsonld/JsonLdCredentialFormatService.ts

```diff
@@ -0,0 +1,440 @@
+    }
+  }
+
+  private compareProofs(credentialProof: LinkedDataProof, requestProof: JsonLdOptionsRFC0593): void {
+    if (credentialProof.domain !== requestProof.domain) {
+      throw new AriesFrameworkError('Received credential proof domain does not match domain from credential request')
+    }
+
+    if (credentialProof.challenge !== requestProof.challenge) {
+      throw new AriesFrameworkError(
+        'Received credential proof challenge does not match challenge from credential request'
+      )
+    }
+
+    if (credentialProof.type !== requestProof.proofType) {
+      throw new AriesFrameworkError('Received credential proof type does not match proof type from credential request')
+    }
+
+    if (credentialProof.proofPurpose !== requestProof.proofPurpose) {
+      throw new AriesFrameworkError(
+        'Received credential proof purpose does not match proof purpose from credential request'
+      )
+    }
+  }
+
+  public supportsFormat(format: string): boolean {
+    const supportedFormats = [JSONLD_VC_DETAIL, JSONLD_VC]
+
+    return supportedFormats.includes(format)
+  }
+
+  public async deleteCredentialById(): Promise<void> {
+    throw new Error('Method not implemented.')
+  }
+
+  public shouldAutoRespondToProposal(
+    agentContext: AgentContext,
+    { offerAttachment, proposalAttachment }: FormatAutoRespondProposalOptions
+  ) {
+    return this.areCredentialsEqual(proposalAttachment, offerAttachment)
+  }
+
+  public shouldAutoRespondToOffer(
+    agentContext: AgentContext,
+    { offerAttachment, proposalAttachment }: FormatAutoRespondOfferOptions
+  ) {
+    return this.areCredentialsEqual(proposalAttachment, offerAttachment)
+  }
+
+  public shouldAutoRespondToRequest(
+    agentContext: AgentContext,
+    { offerAttachment, requestAttachment }: FormatAutoRespondRequestOptions
+  ) {
+    return this.areCredentialsEqual(offerAttachment, requestAttachment)
+  }
+
+  public shouldAutoRespondToCredential(
+    agentContext: AgentContext,
+    { credentialAttachment, requestAttachment }: FormatAutoRespondCredentialOptions
+  ) {
+    // FIXME: we should do a lot of checks to verify if the credential we received is actually the credential
+    // we requested. We can take an example of the ACA-Py implementation:
+    // https://github.com/hyperledger/aries-cloudagent-python/blob/main/aries_cloudagent/protocols/issue_credential/v2_0/formats/ld_proof/handler.py#L492
+    // TODO don't call areCredentialsEqual, call compareStuff() as per aca-py
+
+    const credentialAsJson = credentialAttachment.getDataAsJson<W3cVerifiableCredential>()
+    const credential = JsonTransformer.fromJSON(credentialAsJson, W3cVerifiableCredential)
+
+    const requestAsJson = requestAttachment.getDataAsJson<SignCredentialOptionsRFC0593>()
+    const request = JsonTransformer.fromJSON(requestAsJson, JsonLdCredential)
+
+    if (Array.isArray(credential.proof)) {
```

Comment:
this needs to be reworked

Reply:
Please take a look at the current method: I have made a couple of minor changes; is it ok now? If not, what do we want to change?

Reply:
I now throw an error if it's an array. That should be ok I think.